overlord/snapstate/backend_test.go

```diff
@@ -371,16 +371,19 @@ func (f *fakeStore) SnapAction(ctx context.Context, currentSnaps []*store.Curren
 		if a.Action != "install" && a.Action != "refresh" {
 			panic("not supported")
 		}
+		if a.InstanceName == "" {
+			return nil, fmt.Errorf("internal error: action without instance name")
+		}
 
 		if a.Action == "install" {
-			sspec := snapSpec{
-				Name:     a.Name,
+			spec := snapSpec{
+				Name:     a.InstanceName,
```

Comment:
this doesn't look fully right, at that level we should use just the snap name perhaps.

Reply:
Right, I've some some more changes in this area right here https://github.com/snapcore/snapd/pull/5561/commits/18d926edc343900513101a1fe911c143d01c2cea#diff-147f53d4ad040ca82e13d36e9fdd4e18 but as you suggest, for 'install' case, snap name will be enough. I'll also add eda panic if instance key is set in 'install'

Reply:
Actually I'd leave this as it is now. With later changes from #5561 `f.snap(..)` does the name mangling internally and returns a correct `snap.Info` with instance key and all.

Reply:
that sounds wrong to me, mentally  this bit of the fake store (snap) corresponds to the store so it shouldn't know anything about instance name